### PR TITLE
Add Device Binding tests

### PR DESCRIPTION
**Pull request by vahancouver**
# JIRA Ticket

[SDKS-2189](https://bugster.forgerock.org/jira/browse/SDKS-2189)

# Description

Added tests for the following entities:
DeviceBindingCallback
KeyAware
AuthenticatorFactory
None
BiometricAndDeviceCredential
BiometricOnly
DeviceAuthenticator

# Definition of Done Checklist:

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Ensure backward compatibility.
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] Functional spec is written/updated.
- [ ] Example code snippets have been added.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

### CODE REVIEW

File: FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Callback/DeviceBindingCallbackTests.swift [hunk 1]
```diff
@@ -0,0 +1,2079 @@
...
+//
+
+import XCTest
+import JOSESwift
+@testable import FRAuth
+
+class DeviceBindingCallbackTests: FRAuthBaseTest {
+    
+    func test_01_basic_init() {
+        let jsonStr = """
```
Comment (rodrigoareis) [inline FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Callback/DeviceBindingCallbackTests.swift:18]: We have most of the content of this `jsonStr` are present in all tests. I would suggest move this to a function with parameters to improve the reusability.

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
When checking out the project in XCode, I cannot find the files, but they are present in the filesystem.

**Review by spetrov — APPROVED** [review]
LGTM 👍🏻

**Comment by jeyanthanperiyasamy** [comment]
is there End to End test we can add ?. Like start the journey with providing username and set DeviceBinding to None and verify we get the accessToken/RefreshToken. i think stoyan has a real test server for this..we did this in Android

File: FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Callback/DeviceBindingCallbackTests.swift (hunk 1 above)
Comment (vahancouver) [reply FRAuth/FRAuthTests/FRAuthSwiftTests/FRAuth/Callback/DeviceBindingCallbackTests.swift:41]: @rodrigoareis extracted to a helper method with bunch of parameters with default values. Way less code.

**Pull request merged**